assets/src/domain/eventEditor/services/apollo/mutations/datetimes/useDatetimeMutator.ts

```diff
@@ -1,16 +1,61 @@
+import { useCallback, useMemo } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+
 import { CreateDatetimeInput, UpdateDatetimeInput, DeleteDatetimeInput } from './types';
-import { Datetime } from '@edtrServices/apollo/types';
-import { MutatorFnGn } from '@appServices/apollo/mutations';
-import { useMutator, TypeName } from '../';
+import { MutationType } from '@appServices/apollo/mutations';
+import { CREATE_DATETIME, UPDATE_DATETIME, DELETE_DATETIME } from './';
+import useMutationHandler from './useMutationHandler';
+import useUpdateCallback from '../useUpdateCallback';
+import { MutationFunction, TypeName } from '../types';
 
 interface DatetimeMutator {
-	createEntity: MutatorFnGn<CreateDatetimeInput>;
-	updateEntity: MutatorFnGn<UpdateDatetimeInput>;
-	deleteEntity: MutatorFnGn<DeleteDatetimeInput>;
+	createEntity: MutationFunction<any, CreateDatetimeInput>;
+	updateEntity: MutationFunction<any, UpdateDatetimeInput>;
+	deleteEntity: MutationFunction<any, DeleteDatetimeInput>;
 }
 
 const useDatetimeMutator = (id = ''): DatetimeMutator => {
-	return useMutator<Datetime>(TypeName.Datetime, id) as DatetimeMutator;
+	const [createDatetime] = useMutation(CREATE_DATETIME);
+	const [updateDatetime] = useMutation(UPDATE_DATETIME);
+	const [deleteDatetime] = useMutation(DELETE_DATETIME);
+
+	const mutationHandler = useMutationHandler();
+
+	const getUpdateCallback = useUpdateCallback(TypeName.Datetime);
+
+	const createEntity = useCallback(
+		(input) => {
+			const { onUpdate, ...options } = mutationHandler(MutationType.Create, input);
+
+			const update = getUpdateCallback({ onUpdate, mutationType: MutationType.Create, input });
```

Comment:
nice approach to avoid that issue with calling `useCallback` inside of a function;
it's 2 layers of caching :+1: